frontend/src/components/input-output/add-source-modal/AddSourceModal.jsx

```diff
@@ -98,6 +98,7 @@ function AddSourceModal({
       .catch((err) => {
         setAlertDetails(handleException(err));
         setOpen(false);
+        setEditItemId(null);
```

Comment:
Yes, these are intentional changes. We will set the adapter id to a state when we attempt to edit the adapter. Modal open and API call will be triggered by a change in state. But in this case, if the API call fails, we don't set it back to null. Because there hasn't been a state change, call will not get triggered and user won't receive a warning if they attempt to edit again.